### PR TITLE
brackets: new package, code editor

### PR DESCRIPTION
**Pull request by matejc**

### CODE REVIEW

File: pkgs/applications/editors/brackets/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{ stdenv, fetchurl, buildEnv, gtk, glib, gdk_pixbuf, alsaLib, nss, nspr, gconf
+, cups, libgcrypt_1_5, makeWrapper, dbus, udev }:
+let
+  bracketsEnv = buildEnv {
+    name = "env-brackets";
+    paths = [
+      gtk glib gdk_pixbuf stdenv.cc.cc alsaLib nss nspr gconf cups libgcrypt_1_5
+      dbus udev
+      /* zlib    atk pango freetype libgnome_keyring3
```
Comment (domenkozar) [inline pkgs/applications/editors/brackets/default.nix]: why is this commented? Leave it out if it's not needed

File: pkgs/applications/editors/brackets/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    ar p $src data.tar.xz | tar -C $out -xJ
+
+    mv $out/usr/* $out/
+    rmdir $out/usr
+    ln -sf $out/opt/brackets/brackets $out/bin/brackets
+
+    ln -s ${udev}/lib/libudev.so.1 $out/opt/brackets/lib/libudev.so.0
```
Comment (domenkozar) [inline pkgs/applications/editors/brackets/default.nix]: this looks potentially problematic as it changes ABI from 1 to 0, does it really work?

Comment (matejc) [inline pkgs/applications/editors/brackets/default.nix]: nope, it does not work without it

without that line it returns this error:
```
/nix/store/4ajf4kpv7rwwq0wfafapryap43nd7n56-brackets-1.5/opt/brackets/.brackets-wrapped: error while loading shared libraries: libudev.so.0: cannot open shared object file: No such file or directory
```

and those issues indicate that linking from 1 to 0 is the correct way

https://github.com/adobe/brackets/issues/4720
https://github.com/adobe/brackets/issues/4816
...

File: pkgs/applications/editors/brackets/default.nix (hunk 1 above)
Comment (matejc) [inline pkgs/applications/editors/brackets/default.nix]: done

**Comment by offlinehacker** [comment]
Will this editor improve my css skils?

**Pull request merged**